src/main/java/sparksoniq/spark/udf/ForClauseUDF.java

```diff
@@ -76,11 +79,9 @@ public List call(WrappedArray wrappedParameters) {
 
         DataFrameUtils.deserializeWrappedParameters(wrappedParameters, _deserializedParams, _kryo, _input);
 
-        String[] columnNames = _inputSchema.fieldNames();
-
         // prepare dynamic context
-        for (int columnIndex = 0; columnIndex < columnNames.length; columnIndex++) {
-            _context.addVariableValue(columnNames[columnIndex], _deserializedParams.get(columnIndex));
+        for (int columnIndex = 0; columnIndex < _columnNames.size(); columnIndex++) {
```

Comment:
This code could probably be extracted into a prepareDynamicContext method in a base class since it's used in multiple places.